test/cppsuite/test_harness/workload/thread_context.h

```diff
@@ -52,6 +52,22 @@ class transaction_context {
         return (_in_txn);
     }
 
+    /* Begin a transaction. */
+    void
+    begin(WT_SESSION *session, const std::string &config)
+    {
+        if (!_in_txn) {
+            testutil_check(
+              session->begin_transaction(session, config.empty() ? nullptr : config.c_str()));
+            /* This randomizes the number of operations to be executed in one transaction. */
+            _target_op_count =
+              random_generator::instance().generate_integer<int64_t>(_min_op_count, _max_op_count);
+            op_count = 0;
+            _in_txn = true;
+        } else
+            debug_print("Begin called on a currently running transaction, ignoring.", DEBUG_WARN);
```

Comment:
Do we want to `testutil_die` here? Otherwise, we might be writing tests that are doing this periodically in Evergreen and never notice it which sounds worrisome.

Reply:
3be345076